### PR TITLE
fix: decouple advisory-ID enforcement from tcbStatus check

### LINKED ISSUE #3281: Decouple advisory-ID enforcement from `tcbStatus` check
## Background

Intel announced (2026-05-14, Confidential Computing mailing list) that PCS TCB Info responses will start returning `advisoryIDs` even when `tcbStatus == "UpToDate"`, to convey pertinent information about the platform. Today this is only possible when `tcbStatus != "UpToDate"`.

Our attestation verifier in [`crates/attestation/src/attestation.rs:216-238`](https://github.com/near/mpc/blob/main/crates/attestation/src/attestation.rs#L216-L238) rejects any non-empty `advisory_ids` regardless of `tcbStatus`:

https://github.com/near/mpc/blob/3821d0d74a511f8536f4c0138eed2d95ace57dcd/crates/attestation/src/attestation.rs#L216-L238

Once Intel rolls out the new semantics, currently-valid TDX quotes will start failing with `VerificationError::NonEmptyAdvisoryIds`.

## Acceptance criteria

- `verify_tcb_status` no longer hard-rejects on non-empty advisory IDs when `tcbStatus == "UpToDate"`.
- Decide the policy for the `UpToDate + advisory IDs` case — options:
  - Accept and log/expose the advisory IDs.
  - Accept only if every advisory ID is on an allowlist (overlap with #587).
  - Continue rejecting, but only after explicit triage of each advisory ID Intel starts surfacing.
- Add a regression test using a `VerifiedReport` fixture with `status = "UpToDate"` and non-empty `advisory_ids`.

## Resources

- Intel email: "Intel® Confidential Computing Updates", 2026-05-14
- Related: #587 (which TCB components to enforce)

### PR DESCRIPTION
**Pull request by pbeza**
Closes #3281

## What this fixes

Intel's PCS returns two things about a TEE quote: a **TCB status** (the actual verdict) and an optional list of **`advisory_ids`**. The MPC verifier used to reject *any* non-empty `advisory_ids` as a hard failure — fine when the field only carried real Security Advisories (`INTEL-SA-*`), which Intel only attaches to non-`UpToDate` statuses anyway.

After Intel's 2026 PCS change, an **`UpToDate`** quote can ship with **informational** advisories like `INTEL-DOC-10000` (a lifecycle marker emitted post-ESU). The old code would have rejected those and bricked attestation.

## What changes

**`tcbStatus == UpToDate`** is now the sole security gate. `advisory_ids` become informational output, not a rejection reason.

- `verify_tcb_status` returns advisories as a `Vec<String>` instead of failing on them; `VerificationError::NonEmptyAdvisoryIds` is deleted.
- New `AcceptedDstackAttestation` / `AcceptedAttestation` types bundle the verified attestation with its `advisory_ids` so contract, node, and CLI callers can surface them.
- On acceptance, logs advisory IDs via `env::log_str`, capped at 8 inline entries with a `(+N more)` suffix to keep receipt size bounded against externally-controlled PCS data.
- `crates/attestation-cli` prints advisory IDs under "Informational advisory IDs" on PASS.
- New unit tests cover `UpToDate` ± advisories and `OutOfDate` ± advisories.

## TL;DR

Before: any advisory ID = reject. After: only a non-`UpToDate` TCB status rejects; advisory IDs are logged for visibility.

### CODE REVIEW

File: crates/contract/src/tee/tee_state.rs [hunk 1]
```diff
@@ -152,14 +152,25 @@ impl TeeState {
...
 
+        // Informational advisory IDs (e.g. `INTEL-DOC-10000` after the platform's
+        // Extended Servicing Updates date) may accompany an `UpToDate` TCB status
+        // since Intel's 2026 PCS change. They are not a security failure, but we
+        // log them so operators can see the platform's lifecycle phase in receipts.
+        if !advisory_ids.is_empty() {
+            env::log_str(&format!(
+                "attestation accepted with informational advisory IDs: {}",
+                advisory_ids.join(", ")
+            ));
```
[the remark this answers is not in the document]
Comment (pbeza) [reply crates/contract/src/tee/tee_state.rs]: https://github.com/near/mpc/pull/3296/commits/9f6dda483931b5e56d7a85d85e057a1f833a4541

File: crates/mpc-attestation/src/attestation.rs [hunk 2]
```diff
@@ -132,14 +132,21 @@ pub fn default_measurements() -> &'static [ExpectedMeasurements] {
...
+    /// are expected to log/expose them but should not act on them.
     pub fn verify(
         &self,
         expected_report_data: ReportData,
         current_timestamp_seconds: u64,
         allowed_mpc_docker_image_hashes: &[NodeImageHash],
         allowed_launcher_docker_compose_hashes: &[LauncherDockerComposeHash],
         accepted_measurements: &[ExpectedMeasurements],
-    ) -> Result<VerifiedAttestation, VerificationError> {
+    ) -> Result<(VerifiedAttestation, Vec<alloc::string::String>), VerificationError> {
```
[the remark this answers is not in the document]
Comment (pbeza) [reply crates/mpc-attestation/src/attestation.rs]: https://github.com/near/mpc/pull/3296/commits/9f6dda483931b5e56d7a85d85e057a1f833a4541

**Review by anodar — APPROVED** [review]
Nice!

**Review by netrome — APPROVED** [review]
Looks good, thanks for fixing this 🙏
